core/adapters/http.go

```diff
@@ -169,11 +164,16 @@ func sendRequest(input models.RunInput, request *http.Request, limit int64) mode
 	return models.NewRunOutputCompleteWithResult(responseBody)
 }
 
-func withRetry(client *http.Client, originalRequest *http.Request) (*http.Response, error) {
+func withRetry(
+	client *http.Client,
+	originalRequest *http.Request,
+	timeout time.Duration,
+	attempts uint,
+) (*http.Response, error) {
 	var response *http.Response
 	err := retry.Do(
 		func() error {
-			ctx, cancel := context.WithTimeout(context.Background(), httpDefaultTimeout)
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Did you check into using http.Client's configurable timout? https://golang.org/pkg/net/http/#Client